packages/abstract-signer/src.ts/index.ts

```diff
@@ -199,6 +194,22 @@ export abstract class Signer {
         }
 
         const tx = shallowCopy(transaction);
+        if (!tx.nodeId) {
+            let nodeID : any;
+            if (transaction.nodeId) {
+                nodeID = transaction.nodeId;
+            } else {
+                this._checkProvider();
+                // provider present, we can go on
+                const submittableNodeIDs = this.provider.getHederaNetworkConfig();
+                if (submittableNodeIDs.length > 0) {
+                    nodeID = submittableNodeIDs[0];
+                } else {
+                    logger.throwError("Unable to find submittable node ID. The signer's provider is not connected to any usable network");
+                }
+            }
+            tx.nodeId = nodeID;
+        }
```

Comment:
I think that the `if` logic is wrong -> We should check for `tx.nodeId` and if it is not set, check the provider for it. 
```TypeScript
if (!tx.nodeId) {
    this._checkProvider();
    // provider present, we can go on
    const submittableNodeIDs = this.provider.getHederaNetworkConfig();
    if (submittableNodeIDs.length > 0) {
        tx.nodeId = submittableNodeIDs[0];
    } else {
        logger.throwError("Unable to find submittable node ID. The signer's provider is not connected to any usable network");
    }
}
```